### PR TITLE
[MB-14468] returning correct data types audit history updates

### PR DESCRIPTION
**Pull request by ruizajtruss**
## Summary

When the audit history was updated we were casting all values as strings due to the use of `jsonb_each_text`. That has now been changed to the similar function `jsonb_each` which allows the changed values to maintain their original types. The changes included in this pr are the migrations for the aforementioned change along with small front-end tweaks to adjust for the now properly typed boolean values 🎉 I also created a util function to convert the booleans to 'Yes'/'No' strings for the fields `dependents_authorized` and `has_dependents` in order to better follow the pattern established throughout the rest of the application where similar boolean values are displayed as strings in the same way. Now that the booleans are properly typed, this conversion should no longer be a potential point of failure as it once was. The new util is being used within the `LabeledDetails.jsx` file; I figure it's best to include this generic logic here once rather than copy and pasting it into multiple files adding a lot of extra lines of duplicate code.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Run make db_dev_fresh and ensure the new migrations have been ran
2. Access a move or create a move where `dependents_authorized` or `has_dependents` has been modified and is visible in the move audit history tab
3. If you check the data explorer the field value should now read as a boolean type instead of a string

## Verification Steps for Author

These are to be checked by the author.

- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

## Screenshots
Remains as a boolean when the value is changed:
image

### CODE REVIEW

File: src/utils/formatters.js [hunk 1]
```diff
@@ -277,6 +277,13 @@ export function formatSignatureDate(date) {
   return moment(date).format('YYYY-MM-DD');
 }
 
+// Translate boolean (true/false) into capitalized "Yes"/"No" string
+export const formatYesNoMoveHistoryValue = (value) => {
```
Comment (afoon) [inline src/utils/formatters.js:281]: **Praise**: I thought it was magic when I was seeing `has_dependents: true` show up as 'Yes' in the move history. Great change and use of a util 🙌

File: src/utils/formatters.js [hunk 2]
```diff
@@ -277,6 +277,13 @@ export function formatSignatureDate(date) {
   return moment(date).format('YYYY-MM-DD');
 }
 
+// Translate boolean (true/false) into capitalized "Yes"/"No" string
```
Comment (afoon) [inline src/utils/formatters.js:280]: **Praise**: Was going to suggest a comment but you're already on it 🔥 . Great catch noticing the subtle difference and noting it.

**Review by afoon — APPROVED** [review]
Great work! LGTM. Gnarly migration that works as expected.

**Review by kctruss — APPROVED** [review]
🎉  Fantastic job on this one! You took on a big challenge and nailed it 🔨

**Pull request merged**